src/models/model.cpp

```diff
@@ -500,6 +513,15 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
       for (auto& option : provider_options.options) {
         opts.emplace(option.first, option.second);
       }
+
+      // TODO set device_type_ in a less hacky way.
```

Comment:
I wrote this - any idea on how we can do this better?

Reply:
Yeah, my PR has this function a lot cleaner. When I merge with this, it's really easy to add QnnHtpShared